cf_remote/ssh.py

```diff
@@ -105,12 +119,11 @@ def __exit__(self, *args, **kwargs):
 def connect(host, users=None):
     log.debug("Connecting to '%s'" % host)
     log.debug("users= '%s'" % users)
-    if "@" in host:
-        parts = host.split("@")
-        assert len(parts) == 2
-        host = parts[1]
-        if not users:
-            users = [parts[0]]
+    parts = urlparse("ssh://%s" % host)
+    host = parts.hostname
+    if not users:
+        users = [parts.username]
+    port = parts.port
```

Comment:
What happens if there's no port in `host`? Do we get a `None` here? That would be a problem for multiple things below, AFAICT.

Reply:
```
>>> from urllib.parse import urlparse
>>> o = urlparse("ssh://localhost")
>>> o.port
>>> o
ParseResult(scheme='ssh', netloc='localhost', path='', params='', query='', fragment='')
>>> o.port is None
True
```